From: Egbert Eich <eich@suse.de>
Date: Tue Jan 28 13:23:12 2014 +0100
Subject: drm/mgag200: Reject modes when hdisplay is no multiple of 8
Patch-Mainline: to be upstreamed
Git-commit: 5ed7e77cc1302b1f97a06eb2dac8fc8f68a59b82
Git-repo: git://people.freedesktop.org/~airlied/linux/
References: bnc #806990
Signed-off-by: Egbert Eich <eich@suse.com>

Matrox hardware only supports modes whose horizontal parameters are
multiples of 8. This rules out a mode like 1366x768 for example.

Signed-off-by: Egbert Eich <eich@suse.de>
---
 drivers/gpu/drm/mgag200/mgag200_mode.c |    2 ++
 1 file changed, 2 insertions(+)

--- linux-3.12-SLE12.orig/drivers/gpu/drm/mgag200/mgag200_mode.c
+++ linux-3.12-SLE12/drivers/gpu/drm/mgag200/mgag200_mode.c
@@ -1532,6 +1532,8 @@ static int mga_vga_mode_valid(struct drm
 			bpp) > (55000 * 1024))) {
 		return MODE_BANDWIDTH;
 	}
+	if (mode->hdisplay % 8)
+		return MODE_H_ILLEGAL;
 
 	if (mode->crtc_hdisplay > 2048 || mode->crtc_hsync_start > 4096 ||
 	    mode->crtc_hsync_end > 4096 || mode->crtc_htotal > 4096 ||
